src/Equations/viscoelastic/Model/Setup.cpp

```diff
@@ -74,67 +75,91 @@ void getTransposedViscoelasticCoefficientMatrix( real            i_omega,
   }
 }
 
+template<typename T>
+void getTransposedSourceCoefficientTensor( seissol::model::Material const& material, T& sourceMatrix )
+{
+  sourceMatrix.setZero();
+
+  //       | E_1^T |
+  // E^T = |  ...  |
+  //       | E_L^T |
+  for (unsigned mech = 0; mech < NUMBER_OF_RELAXATION_MECHANISMS; ++mech) {
+    unsigned offset = 9 + mech * 6;
+    real const* theta = material.theta[mech];
```

Comment:
const real* is preferable. (I know your reasoning but imo "everyone does it" is a good reason in this case to prefer the other way)